pkg/types/chain_reader.go

```diff
@@ -44,16 +44,17 @@ type ChainReader interface {
 	// Note that implementations should ignore extra fields in params that are not expected in the call to allow easier
 	// use across chains and contract versions.
 	// Similarly, when using a struct for returnVal, fields in the return value that are not on-chain will not be set.
-	GetLatestValue(ctx context.Context, contractName, method string, params, returnVal any) error
+	GetLatestValue(ctx context.Context, bindingKey string, params, returnVal any) error
 
-	// Bind will override current bindings for the same contract, if one has been set and will return an error if the
-	// contract is not known by the ChainReader, or if the Address is invalid
-	// TODO: (BCF-3139) can't handle log poller filters,explore handling this through chain reader config.
+	// Bind will add provided bindings and will return an error if the contract is not known by the ChainReader, or if
+	// the Address is invalid. Any provided binding that already exists should result in a noop.
 	Bind(ctx context.Context, bindings []BoundContract) error
 
+	// Unbind will remove a binding identified by the binding key and will return an error if the binding is not found.
+	Unbind(ctx context.Context, bindingKey string) error
```

Comment:
nit: If bind is taking multiple contracts, shouldn't unbind allow you to unbind multiple at a time too?

Reply:
Good callout. I'll add that.